server/src/internalClusterTest/java/org/opensearch/remotemigration/MigrationBaseTestCase.java

```diff
@@ -64,4 +83,111 @@ protected void setFailRate(String repoName, int value) throws ExecutionException
+        private Thread indexingThread;
+        private String indexName;
+
+        AsyncIndexingService(String indexName) {
+            this(indexName, Integer.MAX_VALUE);
+        }
+
+        AsyncIndexingService(String indexName, int maxDocs) {
+            indexingThread = new Thread(() -> {
+                while (finished.get() == false && numAutoGenDocs.get() < maxDocs) {
+                    IndexResponse indexResponse = client().prepareIndex(indexName).setId("id").setSource("field", "value").get();
+                    assertEquals(DocWriteResponse.Result.CREATED, indexResponse.getResult());
+                    DeleteResponse deleteResponse = client().prepareDelete("test", "id").get();
+                    assertEquals(DocWriteResponse.Result.DELETED, deleteResponse.getResult());
+                    client().prepareIndex(indexName).setSource("auto", true).get();
+                    numAutoGenDocs.incrementAndGet();
+                    logger.info("Indexed {} docs here", numAutoGenDocs.get());
+                }
+            });
+        }
+
+        public void stopIndexing() throws InterruptedException {
+            finished.set(true);
+            indexingThread.join();
+        }
+
+        public int totalIndexedDocs() {
+            return numAutoGenDocs.get();
+        }
+
+        public void startIndexing() {
+            indexingThread.start();
+        }
+
+        public Thread getIndexingThread() {
+            return indexingThread;
+        }
+    }
+
+    public class SyncIndexingService {
+        private int maxDocs;
+        private int currentIndexedDocs;
+        private boolean forceStop;
+        private String indexName;
+
+        SyncIndexingService(String indexName) {
+            this(indexName, Integer.MAX_VALUE);
+        }
+
+        SyncIndexingService(String indexName, int maxDocs) {
+            this.indexName = indexName;
+            this.maxDocs = maxDocs;
+            this.forceStop = false;
+        }
+
+        public void forceStopIndexing() throws InterruptedException {
+            this.forceStop = true;
+        }
+
+        public int getCurrentIndexedDocs() {
+            return currentIndexedDocs;
+        }
+
+        public void startIndexing() {
+            while (currentIndexedDocs < maxDocs && forceStop == false) {
+                IndexResponse indexResponse = client().prepareIndex(indexName).setId("id").setSource("field", "value").get();
+                assertEquals(DocWriteResponse.Result.CREATED, indexResponse.getResult());
+                DeleteResponse deleteResponse = client().prepareDelete(indexName, "id").get();
+                assertEquals(DocWriteResponse.Result.DELETED, deleteResponse.getResult());
+                client().prepareIndex(indexName).setSource("auto", true).get();
+                currentIndexedDocs += 1;
+                logger.info("Indexed {} docs here", currentIndexedDocs);
+            }
```

Comment:
This is duplicated twice, can we please create this as a method and use this method in the same form in Sync way and for Async you can have a service. Basically, we can get rid of this class and have one method for sync which can be consumed by the async service as well.